src/components/entries/FEEL/FeelPopup.js

```diff
@@ -92,17 +92,34 @@ function FeelPopupComponent(props) {
 
   const editorRef = useRef();
 
+  const isAutoCompletionOpen = useRef(false);
+
   const handleSetReturnFocus = () => {
     sourceElement && sourceElement.focus();
   };
 
-  useEffect(() => {
-    const editor = editorRef.current;
+  const onKeyDownCapture = (event) => {
+
+    // we use capture here to make sure we handle the event before the editor does
+    if (event.key === 'Escape') {
+      isAutoCompletionOpen.current = autoCompletionOpen(event.target);
+    }
+  };
 
-    if (editor) {
-      editor.focus();
+  const onKeyDown = (event) => {
+
+    if (event.key === 'Escape') {
+
+      // close popup only if auto completion is not open
+      // we need to do check this because the editor is not
+      // stop propagating the keydown event
+      // cf. https://discuss.codemirror.net/t/how-can-i-replace-the-default-autocompletion-keymap-v6/3322/5
+      if (!isAutoCompletionOpen.current) {
+        onClose();
+        isAutoCompletionOpen.current = false;
+      }
     }
-  }, [ editorRef, id ]);
+  };
```

Comment:
I thought we could simplify this by checking `event.defaultPrevented` value as it was suggested in https://discuss.codemirror.net/t/how-can-i-replace-the-default-autocompletion-keymap-v6/3322/7
However, a debug session revealed that the library `focus-trap` calls `preventDefault` in between. So we'd need to switch from `preact` event handler to vanilla javascript, which defies the simplification.